core/src/java/show/ezkz/casprzak/core/tools/functiongenerators/TaylorSeries.java

```diff
@@ -0,0 +1,120 @@
+package show.ezkz.casprzak.core.tools.functiongenerators;
+
+import show.ezkz.casprzak.core.config.Settings;
+import show.ezkz.casprzak.core.functions.GeneralFunction;
+import show.ezkz.casprzak.core.functions.binary.Pow;
+import show.ezkz.casprzak.core.functions.commutative.Product;
+import show.ezkz.casprzak.core.functions.commutative.Sum;
+import show.ezkz.casprzak.core.functions.endpoint.Constant;
+import show.ezkz.casprzak.core.functions.endpoint.Variable;
+import show.ezkz.casprzak.core.tools.MiscTools;
+import show.ezkz.casprzak.core.tools.VariableTools;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The methods in {@link TaylorSeries} produce Taylor Series for functions.
+ */
+public class TaylorSeries {
+
+    private TaylorSeries(){}
+
+    /**
+     * Returns a Maclaurin series of the specified function of a given degree
+     * @param function the function whose Maclaurin series is being found
+     * @param degree the degree of the Maclaurin polynomial
+     * @return a Maclaurin series of the specified function
+     */
+    public static GeneralFunction makeTaylorSeries(GeneralFunction function, int degree) {
+        return makeTaylorSeries(function, degree, 0);
+    }
+
+	/**
+	 * Returns a Maclaurin series of the specified function of a given degree
+	 * @param function the function whose Maclaurin series is being found
+	 * @param degree the degree of the Maclaurin polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return a Maclaurin series of the specified function
+	 */
+	public static GeneralFunction makeTaylorSeries(GeneralFunction function, int degree, String variable) {
+		Set<String> variables = VariableTools.getAllVariables(function);
+		if (variables.size() > 1)
+			throw new UnsupportedOperationException("Cannot perform taylor series with a function of more than 1 variable.");
+
+		if (Settings.singleVariableDefault.equals(variable))
+			return makeTaylorSeries(function, degree);
+
+		Map<String, Variable> reverseSubstitution = new HashMap<>() {
+			{
+				put(variable, new Variable(Settings.singleVariableDefault));
+			}
+		};
+		GeneralFunction taylorSeries = makeTaylorSeries(function.substituteVariables(reverseSubstitution), degree);
```

Comment:
Honestly, this seems overly convoluted; why are you substituting it to be a default variable, making the taylor series, then unsubstituting? Wouldn't it be much better to just pass in a Variable to the main taylor series method and have it taylor in that variable? That'd make this whole file much simpler